### ISSUE TITLE
!URGENT!!! - Bot not receiving webhook updates.

### ISSUE THREAD

**Issue by no-nevis** [php-telegram-bot/core#494]
### Required Information

(USING LUMEN - 5.4)
- Operating system: Shared Hosting
- PHP version: 5.6
- PHP Telegram Bot version: 0.42.0
- Using MySQL database: yes
- MySQL version:
- Update Method: Webhook
- Self-signed certificate: no
- RAW update (if available):
```json
{
	"update_id":1111111,
	"message": {
		"message_id":756,
		"from":
		{
			"id":135600675,
			"first_name":"MHOS\u039eI\u0418",
			"last_name":"\u019d",
			"username":"MHN_NJ"
		},
		"chat":
		{
			"id":135600675,
			"first_name":"MHOS\u039eI\u0418",
			"last_name":"\u019d",
			"username":"MHN_NJ",
			"type":"private"
		},
		"date":1494250877,
		"text":"شروع بازی!"
	}
}
```
### Expected behaviour

Telegram handle function should receive webhook updates and process.

### Actual behaviour

handle function never being called at all, in update log i cannot see updates. i even made custom log system to log at very beginning of file, but not receiving anything.
(I think they deliver updates very late - right now i received very old updates)
Webhook info:
`{"url":"https://site.com/11111111/hook","has_custom_certificate":false,"pending_update_count":108,"last_error_date":1494270143,"last_error_message":"Connection timed out","max_connections":40}`

### Steps to reproduce

I dont know.
### Extra details

I'm using Lumen 5.4...
All commands and messages get processed when sending updates manually via Postman. Problem is only about webhook i think.
My Controller:
```php
<?php

namespace App\Http\Controllers;

use Longman\TelegramBot\Entities\WebhookInfo;
use Longman\TelegramBot\Request;
use Longman\TelegramBot\Telegram;
use Longman\TelegramBot\TelegramLog;
use Longman\TelegramBot\Exception\TelegramException;

class HookController extends Controller
{
    protected $telegram;
    protected $api;
    protected $botan;
    protected $bot_name;
    
    public function __construct()
    {
        $this->api = env('TELEGRAM_API');
        $this->bot_name = env('TELEGRAM_USERNAME');
        $this->botan = env('TELEGRAM_BOTAN');
        $this->telegram = new Telegram($this->api, $this->bot_name);
    }

    public function Hook() {
        TelegramLog::initErrorLog(__DIR__ . '/../../../logs/error.log');
        TelegramLog::initDebugLog(__DIR__ . '/../../../logs/debug.log');
        TelegramLog::initUpdateLog(__DIR__ . '/../../../logs/update.log');
        try {
            $this->telegram->enableExternalMySql(app('db')->connection()->getPdo());
            $this->telegram->enableAdmin((int)env('TELEGRAM_ADMIN'));
            if (!empty($this->botan)) $this->telegram->enableBotan($this->botan);
            $this->telegram->addCommandsPath(__DIR__ . '/../../../commands_unused');
            $this->telegram->addCommandsPath(__DIR__ . '/../../../commands');
            $this->telegram->handle();
        } catch (TelegramException $e) {
            TelegramLog::error($e);
        }
    }
    
    public function SetHook() {
        try {
            $result = $this->telegram->setWebhook(route('API.Hook'));
            echo $result->getDescription();
        } catch (TelegramException $e) {
            echo $e;
        }
    }

    public function UnsetHook() {
        try {
            $result = $this->telegram->unsetWebhook();
            echo $result->getDescription();
        } catch (TelegramException $e) {
            echo $e;
        }
    }

    public function WebhookInfo() {
        $result = Request::getWebhookInfo();
        if (!$result->isOk())
            echo 'Webhook was not retrieved! Error: ' . $result->getErrorCode() . ' ' . $result->getDescription();
        else {
            ///** @var WebhookInfo $result */
            //$result = $result->getResult();
            echo $result->toJson();
        }
    }
}

```
Routes file:
```php
<?php

/*
|--------------------------------------------------------------------------
| Application Routes
|--------------------------------------------------------------------------
|
| Here is where you can register all of the routes for an application.
| It is a breeze. Simply tell Lumen the URIs it should respond to
| and give it the Closure to call when that URI is requested.
|
*/

$api = env('TELEGRAM_API');

$app->get('/', function () use ($app) {
    return $app->version();
});

$app->group(
    [
        'prefix' => substr($api, 0, strpos($api, ':')),
        'as' => 'API'
    ],
    function () use ($app) {
        $app->post('hook',
            [
                'as' => 'Hook',
                'uses' => 'HookController@Hook',
            ]);

        $app->get('hook_set',
            [
                'as' => 'SetHook',
                'uses' => 'HookController@SetHook',
        ]);

        $app->get('hook_unset',
            [
                'as' => 'UnsetHook',
                'uses' => 'HookController@UnsetHook',
            ]);

        $app->get('hook_info',
            [
                'as' => 'HookInfo',
                'uses' => 'HookController@WebhookInfo',
            ]);

    }
);
```

**Comment by no-nevis** [comment]
I NEED THIS FIXED VERY SOON. I LOSE MANY IF I DON'T COMPLETE THIS. PLEASE

**Comment by noplanman** [comment]
Hi @no-nevis 

I see that the `pending_update_count` is really high, so something clearly isn't working properly.

Is there anything in your error log?
If you access the URL of your webhook directly, does it add an `Input empty!` message to your error log?

**Comment by no-nevis** [comment]
I don't get ANY errors.
and, no, i get empty response when sending empty request (because my hook route only accept POST :D but even with postman and sending empty post request, i get empty response

**Comment by no-nevis** [comment]
I temporiraly fixed this with switching to getUpdate and cron job every minute, but only temp. (i cant run cli on shared hosting)

**Comment by noplanman** [comment]
Hmm. Did you change anything that might have caused this issue?
Or did it just start happening?

**Comment by no-nevis** [comment]
Just started happening.. i changed some little things long time ago but not remember.

**Comment by noplanman** [comment]
Have you tried resetting your webhook? (unset and set again)

Now that you're using `getUpdates` temporarily, what is the output of your `getWebhookInfo`?

**Comment by no-nevis** [comment]
`{"ok":true,"result":{"url":"","has_custom_certificate":false,"pending_update_count":0}}`

one of my commands was returning nothing, that's fixed but still not working :(

**Comment by noplanman** [comment]
Could you maybe get in touch with me via Telegram (@noplanman)?
Easier to chat and sort it out 😃

**Comment by BrutalBirdie** [comment]
It would be really nice if could at least submit your fix and not just close the Issue after a private chat . . .

**Comment by noplanman** [comment]
@BrutalBirdie Are you having the same problem then?
Please try to reset your API key with BotFather and then reset your webhook.

If that doesn't fix it, feel free to open a new ticket and explain what you've tried and what exactly isn't working, thanks 😃

**Comment by no-nevis** [comment]
@BrutalBirdie It was a problem from the hosting bro, nothing wrong with the bot.

**Comment by jacklul** [comment]
That's the kind of error meaning Telegram has trouble reaching your server.
It might be blocked by firewall or server is located in a country that blocked Telegram completely.
